docs/aqua-book/language/crdt-maps.md

```diff
@@ -0,0 +1,80 @@
+# CRDT Maps
+
+A map is a kind of appendable [collection](types.md#collection-types) that stores key-pairs. A collection of elements can be accessed using keys of type `string`.
+
+How to create a map and write to the map: 
+```aqua
+-- Creation
+map: %u64
+
+-- Append different key-pairs
+map <- "key", 1234
+map <- "key", foo()
+map <- "second key", foo()
+```
+
+Maps can contain only immutable [data types](./types.md#data-types) elements. Arrows, streams, other maps, closures and abilities are not allowed in the type declarations.
+
+Access to keys and elements can create different behaviours based on tasks.
+
+Get array of all elements by key:
+```aqua
+values = map.get("key")
+```
+Get stream of elements by key, later you can join on the stream to wait for specific number of elements in it:
+
+```aqua
+valuesStream <- map.getStream("key")
+-- wait for 9 elements
+join valuesStream[9]
+```
+Get array of all keys (of type `[]string`) in map:
+```aqua
+keys <- map.keys()
+```
+Get stream of keys (of type `*string`):
+```aqua
+keysStream <- map.keysStream
+```
+Check if key contains at least one element, return boolean:
+```aqua
+keyExist <- map.contains("key")
+```
```

Comment:
```suggestion
Get stream of elements by key, later you can join on the stream to wait for specific number of elements in it:

```aqua
valuesStream <- map.getStream("key")
-- wait for 9 elements
join valuesStream[9]
```

Get array of all keys (of type `[]string`) in map:

```aqua
keys <- map.keys()
```

Get stream of keys (of type `*string`):

```aqua
keysStream <- map.keysStream
```

Check if key contains at least one element, return boolean:

```aqua
keyExist <- map.contains("key")
```
```